frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -739,23 +693,22 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
             }),
         ],
 
+        snapshotsLoading: [
+            (s) => [s.snapshotSourcesLoading, s.snapshotsForSourceLoading],
+            (snapshotSourcesLoading, snapshotsForSourceLoading): boolean => {
+                return snapshotSourcesLoading || snapshotsForSourceLoading
+            },
+        ],
+        snapshotsLoaded: [(s) => [s.snapshotSources], (snapshotSources): boolean => !!snapshotSources],
+
         fullyLoaded: [
-            (s) => [
-                s.sessionPlayerSnapshotData,
-                s.sessionPlayerMetaDataLoading,
-                s.sessionPlayerSnapshotDataLoading,
-                s.sessionEventsDataLoading,
-            ],
-            (
-                sessionPlayerSnapshotData,
-                sessionPlayerMetaDataLoading,
-                sessionPlayerSnapshotDataLoading,
-                sessionEventsDataLoading
-            ): boolean => {
+            (s) => [s.snapshots, s.sessionPlayerMetaDataLoading, s.snapshotsLoading, s.sessionEventsDataLoading],
+            (snapshots, sessionPlayerMetaDataLoading, snapshotsLoading, sessionEventsDataLoading): boolean => {
+                // TODO: Do a proper check for all sources having been loaded
                 return (
```

Comment:
Could we check that the number of sources and the snapshots by source counts are the same e.g.
```suggestion
                // TODO: Do a proper check for all sources having been loaded
                return (
                  snapshotSources.length === Object.keys(snapshotsBySource).length
```